tests/test_transform.py

```diff
@@ -37,6 +38,122 @@ def makeRawArrayData(nPoints, nAxes, delta=0.123):
+    @param[in] nAxes  number of axes; ignored if the name is not "Generic"
+    """
+    endpointClassName = name + "Endpoint"
+    endpointClass = getattr(afwGeom, endpointClassName)
+    if name == "Generic":
+        return endpointClass(nAxes)
+    return endpointClass()
+
+
+def makeGoodFrame(name, nAxes):
+    """Return the appropriate frame for the given name and nAzes
+
+    @param[in] name  one of "Generic", "Point2", "Point3" or "SpherePoint"
+    @param[in] nAxes  number of axes; ignored if the name is not "Generic"
+    """
+    return makeEndpoint(name, nAxes).makeFrame()
+
+
+def makeBadFrames(name, nAxes):
+    """Return a list of 0 or more frames that are not a valid match for name and nAxes
+
+    @param[in] name  one of "Generic", "Point2", "Point3" or "SpherePoint"
+    @param[in] nAxes  number of input axes
+    """
+    if name == "Generic":
+        return ()
+    elif name == "Point2":
+        return [
+            astshim.SkyFrame(),
+            astshim.Frame(1),
+            astshim.Frame(3),
+        ]
+    elif name == "Point3":
+        return [
+            astshim.SkyFrame(),
+            astshim.Frame(2),
+            astshim.Frame(4),
+        ]
+    if name == "SpherePoint":
+        return [
+            astshim.Frame(1),
+            astshim.Frame(2),
+            astshim.Frame(3),
+        ]
+    raise RuntimeError("Unrecognized name={}".format(name))
+
+
+def makeFrameSet(baseFrame, currFrame):
+    """Make a FrameSet
+
+    The FrameSet will contain 4 frames and three transforms:
+    baseFrame - UnitMap(nIn) - Frame(nIn) - polyMap - Frame(nOut) - UnitMap(nOut) - currFrame
+
+    where:
+    - nIn = baseFrame.getNaxes()
+    - nOut = currFrame.getNaxes()
+    - polyMap = makePolyMap(nIn, nOut)
+
+    @param[in] baseFrame  base frame; it will be set to ident="baseFrame"
+    @param[in] currFrame  current frame; it will be set to ident="currFrame"
+    """
+    nIn = baseFrame.getNaxes()
+    nOut = currFrame.getNaxes()
+    polyMap = makePolyMap(nIn, nOut)
+
+    baseFrame.setIdent("baseFrame")
+    currFrame.setIdent("currFrame")
+
+    frameSet = astshim.FrameSet(baseFrame)
+    frame2 = astshim.Frame(nIn, "Ident=frame2")
+    frameSet.addFrame(frameSet.getCurrent(), astshim.UnitMap(nIn), frame2)
+    frame3 = astshim.Frame(nOut, "Ident=frame3")
+    frameSet.addFrame(frameSet.getCurrent(), polyMap, frame3)
+    frameSet.addFrame(frameSet.getCurrent(), astshim.UnitMap(nOut), currFrame)
+    frameSet.setIdent("currFrame")
+    return frameSet
+
+
+def permFrameSetIter(frameSet):
+    """Iterator over 0 or more permuted versions of frameSet
+
+    Each returned value is tuple:
+    - permFrameSet: a copy of frameSet with the base and/or current frame permuted
+    - isBasePermuted: a boolean
+    - isCurrPermuted: a boolean
+
+    Only SkyFrames are permuted.
+    Only permuted versions are returned (the original frameSet is never returned)
```

Comment:
These two statements would make more sense immediately after the line about "0 or more permuted versions". Otherwise, the reader is left wondering when it's zero or non-zero.